pkg/go/graph/weighted_graph_node.go

```diff
@@ -1,11 +1,13 @@
 package graph
 
 type WeightedAuthorizationModelNode struct {
-	weights     map[string]int
-	nodeType    NodeType
-	label       string   // e.g. "group#member", UnionOperator, IntersectionOperator, ExclusionOperator
-	uniqueLabel string   // e.g. "group#member", or "union:01JH0MR4H1MBFGVN37E4PRMPM3"
-	wildcards   []string // e.g. "user". This means that from this node there is a path to node user:*
+	weights           map[string]int
+	nodeType          NodeType
+	label             string   // e.g. "group#member", UnionOperator, IntersectionOperator, ExclusionOperator
+	uniqueLabel       string   // e.g. "group#member", or "union:01JH0MR4H1MBFGVN37E4PRMPM3"
+	wildcards         []string // e.g. "user". This means that from this node there is a path to node user:*
+	recursiveRelation string
+	tupleCycle        bool
```

Comment:
This was originally proposed as a map[string]bool where we had the terminal types as the keys because in theory its possible to have a node included in a tuple cycle for the subset of the terminal types assignable for that node. This would offer more precision in identifying which types the cycle applies to as opposed to what I have here which is just saying that there is a cycle for at least one of the types.

I'm deciding to not implement this more precise indicator because the consumers of the weighted graph don't need that information yet and also because this is conceptually simpler. But it could be added later.